### PR TITLE
[Keychron K3] Fix Capslock and OS switch + optical

### LINKED ISSUE #177: [Feature Request] Support Caps lock LED on Keychron K3
The red Caps lock indicator LED does not work on the Keychron K3 (v1, Optical RGB). Instead of turning on/off when capslock is pressed, it is always dimly on.

## Feature Request Type

- [ ] Core functionality
- [ ] Add-on hardware support (eg. audio, RGB, OLED screen, etc.)
- [x] Alteration (enhancement/optimization) of existing feature(s)
- [ ] New behavior

## Description

The build for the RGB Keychron K3 with optical switches works if the `OPTICAL_MATRIX` macro is defined, but the Caps lock indicator feature does not seem to be working. The LED is expected to turn on if Caps lock is enabled and off if Caps lock is disabled. Instead,  the LED is always in a dim-on state, which could indicate a floating gate condition.

PCB version: Ver 1.4/20201030

Built from 03c781d3ca5b024ec3c21d85ad5f323aff70d842 with arm-none-eabi-gcc version 10.1.0 on Windows 10 with QMK MSYS 1.5.2.

### PR DESCRIPTION
**Pull request by seppestas**
Adds support for the Caps lock indicator LED and fix the OS switch. Adds build config for optical version.

## Description

* Fixes #177 by defining `LED_CAPS_LOCK_PIN`

* Fixes the OS selection switch

* Add a rule for the optical version of the Keychron K3 and rename the default keymap to ansi.

* Restore the insert key on fn+del

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 177

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/keychron/k3/info.json [hunk 1]
```diff
@@ -31,7 +31,7 @@
                 "label": "F4",
                 "x": 4,
                 "y": 0
-            },
+            },
```
Comment (dexter93) [inline keyboards/keychron/k3/info.json:34]: whitespace

Comment (seppestas) [reply keyboards/keychron/k3/info.json:34]: Yeah, looks like VS code automatically removed a trailing whitespace here.

File: keyboards/keychron/k3/keymaps/ansi/keymap.c [hunk 2]
```diff
@@ -80,21 +80,21 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
...
-        // do stuff
+      else {
+        print("Bluetooth disabled\n");
       }
       break;
-    case 1: // Connection switch
-      // Probably it's not possible to do anything sensible here as switching from Cable to BT requires turning off the board. (BT / OFF / Cable)
-      if (active) { // BT mode
-        // do stuff
+
```
Comment (dexter93) [inline keyboards/keychron/k3/keymaps/ansi/keymap.c:91]: whitespace again

File: keyboards/keychron/k3/optical/rules.mk [hunk 3]
```diff
@@ -0,0 +1 @@
+OPT_DEFS += -DOPTICAL_MATRIX
```
Comment (dexter93) [inline keyboards/keychron/k3/optical/rules.mk:1]: ☒ Missing keyboards/keychron/k3/optical/readme.md
qmk linter required a readme here too

Comment (seppestas) [reply keyboards/keychron/k3/optical/rules.mk:1]: oh, thanks. Will fix.

**Comment by kriansa** [comment]
Doesn't https://github.com/SonixQMK/qmk_firmware/pull/199 fix it?

**Pull request closed without merging**